tools/rosnode/src/rosnode/__init__.py

```diff
@@ -336,6 +336,9 @@ def rosnode_ping(node_name, max_count=None, verbose=False):
                 if verbose:
                     print("xmlrpc reply from %s\ttime=%fms"%(node_api, dur))
                 # 1s between pings
+            except socket.timeout:
+                print("connection to [%s] timed out"%node_name, file=sys.stderr)
+                return False
```

Comment:
With this extra block added can line [364](https://github.com/ros/ros_comm/pull/1517/files#diff-548698d13d27ce06b14e7ba0e99f6f76R364) still happen?

Reply:
Looking into documentation of [socket](https://docs.python.org/2/library/socket.html) I believe it should not happen when we handle timeout error explicitly.

Socket has 4 exceptions:
- socket.error - can return either a string or (errno, string)
- socket.herror - (h_errno, string)
- socket.gaierror - (error, string)
- socket.timeout - string

since the timeout is the only subclass of socket.error that returns a single string I think the line 364 should never happen (unless I'm missing anything else that could raise a ValueError that is not related to unpacking the error). 

Would you like me to remove this ValueError check?

Reply:
> since the timeout is the only subclass of socket.error that returns a single string I think the line 364 should never happen

The the following message should be changed to state that something "unknown" has happened, e.g.:

```
                     else:
                         print("connection to [%s] timed out"%node_name, file=sys.stderr)
```

replaced by:

```
                     else:
                         print("connection to [%s] failed"%node_name, file=sys.stderr)
```

> Would you like me to remove this ValueError check?

I would rather keep that - there is no reason related to the issue here (the timeout) that it should be removed.

Reply:
Thanks for clarification. I just pushed the change.